tools/bumper/cnaoRepoCommands.go

```diff
@@ -0,0 +1,168 @@
+	return updateComponentsYaml(configPath, componentsConfig)
+}
+
+func (cnaoRepoOps *gitCnaoRepo) bumpComponent(componentName string) error {
+	logger.Printf("Running bump-%s script", componentName)
+	cmd := exec.Command("make", "-C", cnaoRepoOps.gitRepo.localDir, fmt.Sprintf("bump-%s", componentName))
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to run bump script, \nStdout:\n%s\nStderr:\n%s", cmd.Stdout, cmd.Stderr)
+	}
+	return nil
+}
+
+func (cnaoRepoOps *gitCnaoRepo) isComponentBumpNeeded(currentReleaseVersion, latestReleaseVersion, updatePolicy, proposedPrTitle string) (bool, error) {
+	logger.Printf("currentReleaseVersion: %s, latestReleaseVersion: %s, updatePolicy: %s\n", currentReleaseVersion, latestReleaseVersion, updatePolicy)
+
+	if updatePolicy == updatePolicyStatic {
+		logger.Printf("updatePolicy is static. avoiding auto bump")
+		return false, nil
+	}
+
+	// check if PR not already opened
+	isAlreadyOpened, err := cnaoRepoOps.isPrAlreadyOpened(cnaoRepoOps.getCnaoRepoOwnerFromUrl(), cnaoRepoOps.getCnaoRepoNameFromUrl(), proposedPrTitle)
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to check if PR already open")
+	}
+	if isAlreadyOpened {
+		logger.Printf("Bump PR for the latest version already exist. Aborting auto bump")
+		return false, nil
+	}
+
+	// if one of the tags is in vtag format (e.g 0.39.0-32-g1fcbe815), and not equal, then always bump
+	if isVtagFormat(currentReleaseVersion) || isVtagFormat(latestReleaseVersion) {
+		return currentReleaseVersion == latestReleaseVersion, nil
+	}
+
+	currentVersion, err := canonicalizeVersion(currentReleaseVersion)
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to digest current Version %s to semver", currentVersion)
+	}
+	latestVersion, err := canonicalizeVersion(latestReleaseVersion)
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to digest latest Version %s to semver", latestVersion)
+	}
+
+	return currentVersion.LessThan(*latestVersion), nil
+}
+
+// check if there is an already open PR with the same title in the repo.
+func (cnaoRepoOps *gitCnaoRepo) isPrAlreadyOpened(owner, repo, proposedPrTitle string) (bool, error) {
+	logger.Printf("checking if there is an already open bump PR for this release")
+
+	prList, _, err := cnaoRepoOps.githubInterface.ListPullRequests(owner, repo)
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to get list of PRs from %s/%s repo", owner, repo)
+	}
+
+	for _, pr := range prList {
+		if pr.GetTitle() == proposedPrTitle {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (cnaoRepoOps *gitCnaoRepo) getCnaoRepoNameFromUrl() string {
+	urlSlice := strings.Split(cnaoRepoOps.configParams.Url, "/")
+	return urlSlice[len(urlSlice)-1]
+}
+
+func (cnaoRepoOps *gitCnaoRepo) getCnaoRepoOwnerFromUrl() string {
+	urlSlice := strings.Split(cnaoRepoOps.configParams.Url, "/")
+	return urlSlice[len(urlSlice)-2]
+}
```

Comment:
They are only use for `isPrAlradyOpenned` so we can just do the split once there and access the position.

Reply:
in future PRs I'm going to add more functions that use this so it's better to keep it.